### ISSUE TITLE
Allowing multiple licenses per upload

### ISSUE THREAD

**Issue by frankois** [zenodo/zenodo#2026]
This could be useful for example to publish software and documentation under different licenses while uploading it together.

**Comment by logological** [comment]
To expand on this, allowing multiple licences per upload would also cover the case that a software package combines source code with different but [compatible](https://en.wikipedia.org/wiki/License_compatibility) licences.  For example, it's quite common to see software where the primary authors have released their contributions under the [GNU General Public License](https://www.gnu.org/licenses/gpl.html), but which includes (possibly modified) third-party code released under the [modified BSD License](https://directory.fsf.org/wiki/License:BSD-3-Clause).

It's also not unheard of for a software package to include files with incompatible (but still free/open-source) licences.  Probably the most common case, as @frankois mentions, is when a package bundles software and its associated documentation: the software may be released under a free/open-source licence whereas the documentation may be released under a free content licence (since software licences aren't really appropriate for documentation).  But it's also not unheard of for a package to contain incompatibly licensed programs that are designed to legally interoperate in some manner. (In fact, the FSF even has [an entire FAQ entry](https://www.gnu.org/licenses/gpl-faq.en.html#GPLIncompatibleLibs) that covers the case of linking GPL'd code to incompatible libraries.)

**Comment by nemobis** [comment]
I agree this is useful for dual-licensed projects and works, less so for other use cases.

> the case that a software package combines source code with different but compatible licences

Note that this is one reason you're often supposed to have per-file license headers. Neverending discussion, but see e.g. https://lu.is/blog/2012/03/17/on-the-importance-of-per-file-license-information/ for a glimpse of it.

**Comment by iishiishii** [comment]
any updates on this feature?
